server/text_generation_server/models/custom_modeling/flash_gpt2_modeling.py

```diff
@@ -106,7 +106,7 @@ def _load_qkv(config, prefix: str, weights, head_size, num_heads):
     for i in range(3):
         tensor = slice_[:, start + i * single_size : stop + i * single_size]
         tensors.append(tensor)
-    weight = torch.cat(tensors, dim=1).T
+    weight = torch.cat(tensors, dim=1).T.contiguous()
```

Comment:
These non-contiguous weights were causing wrong generation on ROCm with LLMM1 that has no check on input tensors contiguity.